src/utils/directoryListing.ts

```diff
@@ -0,0 +1,123 @@
+import Handlebars from 'handlebars';
+import type { File, ReadDirectoryResult } from '../providers/provider';
+import htmlTemplate from '../templates/directoryListing.out';
+import { toReadableBytes } from '../utils/object';
+
+const handlebarsTemplate = Handlebars.template(htmlTemplate);
+
+const months = [
+  'Jan',
+  'Feb',
+  'Mar',
+  'Apr',
+  'May',
+  'Jun',
+  'Jul',
+  'Aug',
+  'Sep',
+  'Oct',
+  'Nov',
+  'Dec',
+];
+
+/**
+ * Render a directory listing
+ *
+ * TODO: Work with other teams on removing their dependency on nginx's listing
+ *  result so we don't need to emulate it
+ */
+export function renderDirectoryListing(
+  url: URL,
+  result: ReadDirectoryResult
+): string {
+  const tableElements: TableElement[] = [];
+
+  for (const name of result.subdirectories) {
+    tableElements.push(renderSubdirectory(name));
+  }
+
+  // Last time the directory was "modified", just the last modified of the most
+  //  recently updated file
+  let directoryLastModified: Date | undefined;
```

Comment:
```suggestion
  let directoryLastModified: Date;
```

Implict that default value is undefined